src/core/services/aggregation.js

```diff
@@ -81,15 +81,20 @@ export default class Aggregation {
 		return [min, max];
 	}
 
-	static avg(rows, getValue) {
+	static avg(rows, getValue, aggregationOptions) {
 		if (!rows.length) {
 			return null;
 		}
 
+		if (aggregationOptions.distinct) {
+			let uniqueSet = new Set();
+			return Aggregation.sum(rows, getValue, aggregationOptions, uniqueSet) / uniqueSet.size;
+		}
+
 		return Aggregation.sum(rows, getValue) / rows.length;
 	}
 
-	static sum(rows, getValue) {
+	static sum(rows, getValue, aggregationOptions, externalSet) {
```

Comment:
This is the Set for filling from another method. I use it in AVG for dividing sum on it's length